### DISCUSSION TITLE
Include an app to have a better clipboard by default

### DISCUSSION THREAD

**Discussion by spotlesscoder** [elementary/triage#624] [Ideas]
No idea where else to file this - feel free to move it

Situation:
I was writing a review for a vscode extension on vs code market place
When I wanted to submit the review, I was already expecting to be redirected to the sign in page first and potentially having to retype my review after login. So I CTRL + C'd my review to the clipboard

When signing in I had to copy-paste my one-time login code from the email microsoft sent me
I didn't think about it and also CTRL + C'd that code

So my review in the clipboard was gone

I want to access a history of like at least 5-10 clipboard entries in every elementary installation without having to manually setup or install anything
For security reasons, older entries might be deleted after a few minutes (making sure no passwords stay in there for too long), only keeping the latest copied text "forever" (current session)

**Comment by teamcons** [comment]
A case can be made for this, definitively.
Windows has a history appearing on CTRL+V. It is pretty handy, and always there - Once in a while it is needed when juggling with copied entries, or recovering a lost entry.
I need to look whether i can do something like this.

**Comment by diogopessoa** [comment]
A clipboard is super necessary. For writing code, editing text. At the moment I'm using Copyq to cover this absence.

**Reply by teamcons** [reply]
How open would you be to help me on this ? I see you code, and have a vala repo - Im unfortunately not a vala expert so i would be slow at best

there is at the time a rudimentary-and-ugly wingpanel-indicator which does this. It works, but is as basic as it gets

i would like to do a windows-style clipboard: starts in the background at start logging entries, shows a history window when called

**Reply by diogopessoa** [reply]
Sorry, but I still don't program. I forked a GitHub repository just out of curiosity and to try to learn. Sorry I can't help you 😔

**Reply by diogopessoa** [reply]
@teamcons There's a discussion about the Clipboard on Discord. I think you'd like to take part: 

https://discord.com/channels/611717943788699724/879476008548454511

**Reply by teamcons** [reply]
> @teamcons There's a discussion about the Clipboard on Discord. I think you'd like to take part:

Yes, i am part of it. I am one of the people driving it, suggesting ideas, it was my concept art being shared and me sharing jeremy's draft :D
Im Stella. Hello !

**Reply by diogopessoa** [reply]
Nice! Glad to hear it. 😉

**Comment by 1280px** [comment]
There used to be a "Clippy" applet in Plank prior to its replacement with Dock in OS 8, pretty basic but still handy and right out of the box.

So, if you don't mind using X11 instead of Wayland (or if you're running OS 7.1 or lower — which you probably are), you can re-install Plank, add Clippy from applets in Plank settings and have a clipboard manager applet right on your dock.

**Reply by doobes** [reply]
Well, 

I'm on the crux of switching to X11. There have been just way to many regressions with the transition to Wayland.

I'm also finding several of the apps I use don't behave nicely under Wayland.

Just yesterday, I started an instance of XnView MP. When I closed it, the app closed, but the image of XnView did not. The image was in front of all the other applications. The only recourse was to reboot EoS.

I've got a v7.1 box I was going to upgrade. That's not likely to happen now.

**Reply by teamcons** [reply]
> Well,
> 
> I'm on the crux of switching to X11. There have been just way to many regressions with the transition to Wayland.
> 
> I'm also finding several of the apps I use don't behave nicely under Wayland.
> 
> Just yesterday, I started an instance of XnView MP. When I closed it, the app closed, but the image of XnView did not. The image was in front of all the other applications. The only recourse was to reboot EoS.
> 
> I've got a v7.1 box I was going to upgrade. That's not likely to happen now.

ok
